### PR TITLE
Fix replication_slots schema

### PR DESCRIPTION
**Pull request by mikewallace1979**
Fixes a bug introduced when making replication_slots nullable by reverting the change and making the ReplicationSlot type nullable directly.

### CODE REVIEW

File: pg_backup_api/pg_backup_api/openapi_server/models/status.py [hunk 1]
```diff
@@ -1162,20 +1162,30 @@ def postgresql_systemid(self, postgresql_systemid):
...
         """Sets the replication_slot of this Status.
 
+        Details of the replication slot being used by Barman on this PostgreSQL server in the form of an ordered array. The first item MUST be a string containing the slot_name. The second item MUST be a boolean reflecting the active state of the replication slot. The third item MUST be a string containing the LSN of the oldest WAL which still might be required by the consumer of this slot.   # noqa: E501
 
         :param replication_slot: The replication_slot of this Status.
-        :type replication_slot: List
-        """
+        :type replication_slot: List[object]
+        """
+        if replication_slot is not None and len(replication_slot) > 3:
```
Comment (didiermichel) [inline pg_backup_api/pg_backup_api/openapi_server/models/status.py:1181]: Why use  two conditions and not test  `len(replication_slot) == 3` and display both expected and actual number of elements in error message?

Comment (didiermichel) [reply pg_backup_api/pg_backup_api/openapi_server/models/status.py:1181]: Just wondering: are those 3 elements prone to error? ie do we need to test them (string, bool, string) and maybe more complex test?

Comment (mikewallace1979) [reply pg_backup_api/pg_backup_api/openapi_server/models/status.py:1181]: Good questions.

This is openapi-generated code and it's verifying that the data matches the schema. There's no reason why the we *would* need more testing - in practice either all elements will be there or it will be null.

We could improve it but I don't want to have to dig too deeply into openapi/connexion at this point.

Comment (didiermichel) [reply pg_backup_api/pg_backup_api/openapi_server/models/status.py:1181]: Well if is is an automation no need to fight agains it (someone has probably already tested this).

**Review by didiermichel — APPROVED** [review]
